src/Core/src/Test/ResultMockFactory.php

```diff
@@ -89,7 +89,27 @@ public static function create(string $class, array $data = [])
 
         $initializedProperty->setValue($object, true);
         foreach ($data as $propertyName => $propertyValue) {
-            $property = $reflectionClass->getProperty($propertyName);
+            if ($reflectionClass->hasProperty($propertyName)) {
+                // backward compatibility with `UpperCamelCase` naming
+                $property = $reflectionClass->getProperty($propertyName);
+            } elseif ($reflectionClass->hasProperty(\lcfirst($propertyName))) {
+                // backward compatibility with `UpperCamelCase` naming
+                $property = $reflectionClass->getProperty(\lcfirst($propertyName));
+            } else {
+                $lowerPropertyName = \strtolower($propertyName);
```

Comment:
I just moved down the comment. 

```suggestion
            if ($reflectionClass->hasProperty($propertyName)) {
                $property = $reflectionClass->getProperty($propertyName);
            } elseif ($reflectionClass->hasProperty(\lcfirst($propertyName))) {
                // backward compatibility with `UpperCamelCase` naming (fast)
                $property = $reflectionClass->getProperty(\lcfirst($propertyName));
            } else {
                // backward compatibility with `UpperCamelCase` naming (slow)
                $lowerPropertyName = \strtolower($propertyName);
```

Reply:
This is not totally accurate. I reword the 3 coments